sw/device/silicon_creator/lib/ownership/isfb.c

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/flash_ctrl.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/manifest.h"
+#include "sw/device/silicon_creator/lib/ownership/owner_block.h"
+
+#include "flash_ctrl_regs.h"
+
+enum {
+  kFlashPageBytes = FLASH_CTRL_PARAM_BYTES_PER_PAGE,
+  kFlashPageWords = kFlashPageBytes / sizeof(uint32_t),
+
+  // The maximum number of product expressions is 256, which maps to 512 flash
+  // words or 1024 bytes.
+  kProductExprMaxCount = 256,
+
+  // The maximum size of the product expressions in bytes.
+  KProductExprMaxCountBytes = kProductExprMaxCount * sizeof(uint32_t),
+
+  // The number of bits in the strike mask is fixed to 128. Each bit in the
+  // strike mask corresponds to a flash word.
+  kExpectedStrikeBitCount = 128,
+
+  // The expected size of the strike region in bytes and words.
+  kExpectedStrikeRegionBytesCount = kExpectedStrikeBitCount * sizeof(uint32_t),
+  kExpectedStrikeRegionWordsCount =
+      kExpectedStrikeRegionBytesCount / sizeof(uint32_t),
+
+  // The layout of the ISFB info page is as follows:
+  //
+  // | byte start      | byte end | description
+  // |-----------------|----------|------------
+  // | 0               | 511      | Strike mask
+  // | 512             | 1023     | Unused and available to owner firmware.
+  // | 1024            | OnwerCfg | Product expression constraints.
+  // | 1024 + OwnerCfg | 2047     | Unused and available to owner firmware.
+  kIsfbWordOffsetStartStrikeRegion = 0,
+  kIsfbWordOffsetStartProductExpr = 1024 / sizeof(uint32_t),
+};
+static_assert(kExpectedStrikeRegionWordsCount == kExpectedStrikeBitCount,
+              "Unexpected strike region size");
+static_assert(kIsfbWordOffsetStartProductExpr * sizeof(uint32_t) +
+                      KProductExprMaxCountBytes ==
+                  kFlashPageBytes,
+              "Product expression total size overflows page size");
+
+rom_error_t isfb_boot_request_process(const manifest_ext_isfb_t *ext,
+                                      const owner_config_t *owner_config,
+                                      uint32_t *checks_performed_count) {
+  *checks_performed_count = UINT32_MAX;
+  if (ext->header.name != kManifestExtIdIsfb ||
+      (hardened_bool_t)owner_config->isfb == kHardenedBoolFalse) {
+    return kErrorOwnershipISFBNotPresent;
+  }
+  const owner_isfb_config_t *isfb = owner_config->isfb;
+  if (ext->product_expr_count * 2 > isfb->product_words ||
+      ext->product_expr_count > kProductExprMaxCount) {
+    return kErrorOwnershipISFBProductExpCnt;
+  }
+
+  flash_ctrl_perms_t perm = {
+      .read = kMultiBitBool4True,
+      .write = kMultiBitBool4True,
+      .erase = kMultiBitBool4False,
+  };
+  flash_ctrl_cfg_t cfg = {
+      .scrambling = kMultiBitBool4False,
+      .ecc = kMultiBitBool4True,
+      .he = kMultiBitBool4False,
+  };
```

Comment:
Do we need to configure this here?  I'd imagined that the normal owner FlashInfoConfig block would handle the info page properties wanted.

Reply:
Removed. I will take care of orchestrating the right  order of operations when I connect this in the `rom_ext`.